merlin/table/tensor_table.py

```diff
@@ -145,6 +155,12 @@ def columns(self):
         """
         return list(self.keys())
 
+    @columns.setter
+    def columns(self, col_names):
+        for col, col_name in zip(self.columns, col_names):
+            self._columns[col_name] = self._columns[col]
+            del self._columns[col]
```

Comment:
there could be an edge case here where a column is being renamed to an already existing column, which might result in missing columns in the result? e.g. one simple test case might be to swap labels between two columns